app/controllers/application_controller.rb

```diff
@@ -86,11 +86,16 @@ def render_unauthorized
   end
 
   def saml_settings
-    settings = SAML::SettingsService.new.saml_settings
+    if defined?(@saml_settings)
+      @saml_settings.name_identifier_value = @session&.uuid if defined?(@session)
```

Comment:
I think the `if defined?(@session)` is redundant when you're already using the safe navigation operator (`&`).